runelite-client/src/main/java/net/runelite/client/plugins/StatusBars/StatusBarsOverlay.java

```diff
@@ -0,0 +1,300 @@
+				renderBar(g, offsetPrayerX, offsetPrayerY ,
+					maxPrayer, currentPrayer, WIDTH, height,
+					PADDING, prayerBar, prayerImage.getScaledInstance(imageSize , imageSize , Image.SCALE_SMOOTH), counter , counterPrayerText , counterPrayerX);
+
+				// gets the value from ItemStat plugin to build the hp or prayer bar accordingly to the value of the item a player hovers over.
+				WidgetItem[] inventory = queryRunner.runQuery(new InventoryWidgetItemQuery());
+				java.awt.Point mousePos = new java.awt.Point(client.getMouseCanvasPosition().getX(), client.getMouseCanvasPosition().getY());
+				for (WidgetItem item : inventory)
+				{
+					if (item.getCanvasBounds().contains(mousePos))
+					{
+						Effect change = statChanges.get(item.getId());
+						if (change != null)
+						{
+							StringBuilder b = new StringBuilder();
+							StatsChanges statsChanges = change.calculate(client);
+							for (StatChange c : statsChanges.getStatChanges())
+							{
+								if (c.getStat().getName() == "Hitpoints")
+								{
+									foodHealValue = Integer.parseInt(c.getTheoretical());
+								}
+
+								if (c.getStat().getName() == "Prayer")
+								{
+									prayerHealValue = Integer.parseInt(c.getTheoretical());
+								}
+							}
+						}
+					}
+				}
+				// Bar for healing health or payer points
+				renderHealingBar(g, offsetHealthX , offsetHealthY,
+						maxHealth, currentHealth, WIDTH , height,
+						PADDING, foodHealValue, HEAL_COLOR);
+				renderHealingBar(g, offsetPrayerX , offsetPrayerY,
+						maxPrayer, currentPrayer, WIDTH, height,
+						PADDING, prayerHealValue, PRAYER_HEAL_COLOR);
+			}
+		}
+		return null;
+	}
+
+	private void renderBar(Graphics2D graphics, int x, int y, int max, int current, int width, int height, int padding, Color filled, Image image, Color counter_color, String counter, int counter_x)
+	{
+		//draw background
+		graphics.setColor(BACKGROUND);
+		graphics.fillRect(x, y, width, height);
+
+		//draw bar background
+		graphics.setColor(DEPLETED);
+		graphics.fillRect(x + padding, y + padding, width - padding * 2, height - padding * 2);
+
+		//draw bar with current health or prayer points
+		int filledHeight = getBarHeight(max, current, height);
+		graphics.setColor(filled);
+		graphics.fillRect(x + padding, y + padding + (height - filledHeight), width - padding * 2, filledHeight - padding * 2);
+
+		//draw counter
+		graphics.setColor(counter_color);
+		graphics.setFont(FontManager.getRunescapeSmallFont());
+
+		if (config.enableCounter() && !config.enableSkillIcon())
+		{
+			graphics.drawString(counter, x + counter_x  + padding, y + 20);
+		}
+		else
+		{
+			graphics.drawString(counter, x + counter_x + padding, y + 35);
+		}
+
+		//draw icons
+		graphics.drawImage(image, x + 1 + padding, y + 21 - image.getWidth(null), null);
+	}
+
+	private void renderHealingBar(Graphics2D graphics, int x, int y, int max, int current, int width, int height, int padding, int heal, Color color)
```

Comment:
This can be static

Reply:
some things cannot be refferenced from a static context i get if i do that.